src/java.base/share/classes/java/security/Provider.java

```diff
@@ -1604,6 +1604,8 @@ private static void addEngine(String name, boolean sp, Class<?> constructorParam
         addEngine("KeyGenerator",                       false, null);
         addEngine("SecretKeyFactory",                   false, null);
         addEngine("KEM",                                true,  null);
+        addEngine("KDF",                                true,
```

Comment:
For the 2nd argument to be true, the HKDF provider implementation should probably register some additional service attributes as defined in https://docs.oracle.com/en/java/javase/22/docs/specs/security/standard-names.html#service-attributes